### DISCUSSION TITLE
Inconsistent Search Results for Tagged Metadata in PnP Search Web Parts

### DISCUSSION THREAD

**Discussion by Tishaay** [microsoft-search/pnp-modern-search#4015] [Q&A]
We are utilizing the PnP Search Results and PnP Search Bar web parts on our Modern SharePoint site. Our document library includes metatags, and the corresponding crawl properties have been properly mapped to their respective managed properties. We have configured our search results to filter and display documents based on specific metatags. While the search functionality works as expected for some keywords/metatags, allowing us to see the files with the associated tagged metadata, we have noticed inconsistencies during testing. Certain keywords/metatags do not return any results, despite the fact that the documents in the library have been appropriately tagged. 

Please refer to the attached screenshot for further details.

We have tagged several files with the "HBL" metadata, and when searched, the results correctly display those tagged files:
![image](https://github.com/user-attachments/assets/9ea2c13c-7534-4ac7-8bc2-a8fe01accee8)
![image](https://github.com/user-attachments/assets/479baa4d-424a-4f24-929a-45073113ecbf)

However, for files tagged with the "Intimate Care" metadata, no results are returned when searching for this keyword, despite the fact that the files are correctly tagged:
![image](https://github.com/user-attachments/assets/057ffa65-5039-44cc-8cfa-1afb00519c91)
![image](https://github.com/user-attachments/assets/f76524d0-234c-4fa6-95e7-71526eeded58)

Could you help identify what might be causing this discrepancy?

**Comment by wobba** [comment]
Curious if you "broke" search due to what I write about here: https://www.techmikael.com/2020/03/side-effect-of-mapping-crawled.html

**Reply by Tishaay** [reply]
This issue is already addressed. However, I am experiencing another issue. Specifically, I have metatags that include special characters, and this seems to be causing problems with search functionality. When I search using the metatag with the special character, no results are returned. However, if I search without the special character, the file associated with that metatag does appear. 

For example, I have metatag like "Pond's". When I search for "Pond's" (with the apostrophe) and "Ponds" (without the apostrophe), no results are shown. But when I search for "Pond" (without the special character), the correct file is displayed.

How should I configure my search query template to ensure that it returns results regardless of whether a user searches for "Pond's" (with the apostrophe), "Ponds" (without the apostrophe), or simply "Pond" (without any special characters)? I would like the search to recognize all variations and display relevant results for each query.

Here is my current query template
![image](https://github.com/user-attachments/assets/2c81334b-cc69-4dc1-bb78-3e3e414290bf)

**Reply by wobba** [reply]
Your KQL is incorrect as you have a space between `:` and `"`. This breaks parsing. KQL does not allow any spaces between properties, operators and values.

**Comment by kasperbolarsen** [comment]
What exactly is a "metatag"? Is it a Manage Metadata field , aka a Taxonomy field?
